src/Reflection/ClassReflection.php

```diff
@@ -473,16 +476,28 @@ public function hasProperty(string $propertyName): bool
 
 	public function hasMethod(string $methodName): bool
 	{
+		$key = sprintf('%s-%s', $methodName, $this->getCacheKey());
```

Comment:
This cache key doesn't make sense. Here we are inside a specific ClassReflection instance. The cache property isn't static. I believe this could just be `$methodName;` as it's in `getMethod`. In there the `$key` is modified when the `$scope` is in class, but `hasMethod` doesn't have `$scope` parameter so the cache key can simply be `$methodName`.

Remove the `$key` variable altogether, and just use `$methodName` when accessing the array. Thank you.

Reply:
Yep. makes total sense. Updated.